File: PyFlation/prettyDF.py
# Cleans up a dataframe by removing unnecessary columns with useless input
########## Module

def remove_1D_rows(df):
	df = df.copy()
	for col in df.columns:
		column = map(lambda x: str(x), df[col])
		column = set(column)
		if len(column) == 1:
			del df[col]
	return df


compact.df = remove_1D_rows(df)

File: Inflation/prettyDF.r
# Cleans up a dataframe by removing unnecessary columns with useless input
########## Module

remove.1D.cols = function(data) {
	factored.data = lapply(data, factor)
	nleveler = Vectorize(nlevels)
	data[!(nleveler(factored.data) <= 1)]
}

# Get rid of the unnecessary columns (1 dimensional/no information)
compact.df = remove.1D.cols(df)

